### PR TITLE
Add support for multiple stream libraries (Cycle Unified)

### PR DESCRIPTION
**Pull request by lmatteis**

### CODE REVIEW

**Comment by lmatteis** [comment]
Notable changes: 
* made xstream a peerDependency
* need to call `setAdapt` when running rxjs tests (as it sets a global adapter)
* updated README with multiple streaming libs docs

File: README.md [hunk 1]
```diff
@@ -30,10 +30,10 @@ Handle redux async actions using [Cycle.js](https://cycle.js.org/).
 
 `npm install --save redux-cycles`
 
-Then use `createCycleMiddleware()` which returns the redux middleware function, but also has two function properties attached to it; namely `makeActionDriver()` and `makeStateDriver()` which you can use accordingly when you call `Cycle.run` (which can be installed via `npm i -s @cycle/xstream-run`).
+Then use `createCycleMiddleware()` which returns the redux middleware function, but also has two function properties attached to it; namely `makeActionDriver()` and `makeStateDriver()` which you can use accordingly when you call your Cycle `run` function (can be installed via `npm install --save @cycle/run`).
```
Comment (nickbalestra) [inline README.md]: Maybe is a bit too long? What about breaking it up into something like:

Then use `createCycleMiddleware()` which returns the redux middleware function with two driver factories attached: `makeActionDriver()` and `makeStateDriver()`.  Use them when you call the Cycle `run` function (which can be installed via `npm install --save @cycle/run`).

File: README.md [hunk 2]
```diff
@@ -59,6 +59,21 @@ run(main, {
 })
 ```
 
+By default `@cycle/run` uses `xstream`. If you want to use another streaming library in your code simply import them and use their `run` method instead.
+
+For RxJS:
+
+```js
+import { run } from '@cycle/rxjs-run';
```
Comment (nickbalestra) [inline README.md]: Do you think we should add import Rx from ... etc on those examples as well for clarity?

File: example/package.json [hunk 3]
```diff
@@ -23,16 +22,16 @@
...
     "babel-core": "^6.11.4",
     "babel-loader": "^6.2.4",
     "babel-preset-es2015": "^6.9.0",
     "babel-preset-react": "^6.11.1",
-    "@cycle/time": "^0.2.1",
+    "jest": "^18.1.0",
     "webpack": "^1.13.1",
-    "webpack-dev-server": "^1.14.1",
-    "jest": "^18.1.0"
+    "webpack-dev-server": "^1.14.1"
```
Comment (nickbalestra) [inline example/package.json]: love the ordering 💅

File: src/createCycleMiddleware.js [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import xs from 'xstream'
+import {adapt} from '@cycle/run/lib/adapt'
```
Comment (nickbalestra) [inline src/createCycleMiddleware.js]: Should we use `adapt` also on https://github.com/cyclejs-community/redux-cycles/blob/master/src/combineCycles.js#L17 ?

File: test/createCycleMiddleware.test.js [hunk 5]
```diff
@@ -54,9 +60,9 @@ describe('Redux cycle middleware', () => {
       const pong$ = sources.ACTION
         .filter(action => action.type === 'PING')
         .map(() =>
-            xs.periodic(10000)
-                .take(1)
-                .mapTo({ type: 'PONG' })
+          xs.periodic(10000)
```
Comment (nickbalestra) [inline test/createCycleMiddleware.test.js]: I've asked this before, but why not changing this now and importing `delay` from the extra operators?

File: src/createCycleMiddleware.js (hunk 4 above)
Comment (lmatteis) [inline src/createCycleMiddleware.js]: Good point. Will it work though? If the user is using rxjs-run, then the sinks will be RxJS streams, hence `xs.merge(rxjsOsbervable)` will break. How do we make it work?

File: README.md (hunk 2 above)
Comment (lmatteis) [inline README.md]: you don't usually need to import Rx unless you need its stuff though? Which is beyond the scope of `run`.

File: README.md (hunk 1 above)
Comment (lmatteis) [inline README.md]: 👍

File: README.md (hunk 2 above)
Comment (nickbalestra) [inline README.md]: 👍

File: src/createCycleMiddleware.js (hunk 4 above)
Comment (nickbalestra) [inline src/createCycleMiddleware.js]: Saw the gitter 👍

Comment (lmatteis) [inline src/createCycleMiddleware.js]: staltz answered. check my latest commit. didn't even test it but i think it should work, right?

Comment (nickbalestra) [inline src/createCycleMiddleware.js]: 👍 Will add a second PR with more tests for it

Comment (lmatteis) [inline src/createCycleMiddleware.js]: for now i added it in the tests (new commit) to make sure nothing breaks. should be enough to test it there

File: test/createCycleMiddleware.test.js (hunk 5 above)
Comment (Widdershin) [inline test/createCycleMiddleware.test.js]: You could also potentially use `@cycle/time` for these tests if you'd like them to best fast 😄

**Comment by lmatteis** [comment]
@nickbalestra @goshakkk I want to merge this asap. Which version should we go with? @Widdershin might help with versioning as well. Are we ready to go for 1.0.0 ? Or too soon?

**Comment by nickbalestra** [comment]
I would go with 1.x

**Pull request merged**